tests/testsuite/profile_trim_paths.rs

```diff
@@ -57,3 +61,380 @@ Caused by:
+}
+
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn diagnostics_works() {
+    Package::new("bar", "0.0.1")
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { let unused = 0; }"#)
+        .publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = "0.0.1"
+
+                [profile.dev]
+                trim-paths = "diagnostics"
+           "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    let registry_src = paths::home().join(".cargo/registry/src");
+    let registry_src = registry_src.display();
+
+    p.cargo("build -vv -Ztrim-paths")
+        .masquerade_as_nightly_cargo(&["-Ztrim-paths"])
+        .with_stderr_contains(format!(
+            "\
+warning: unused variable: `unused`
+ --> {registry_src}/[..]/bar-0.0.1/src/lib.rs:1:18
+  |
+1 | pub fn f() {{ let unused = 0; }}
+  |                  ^^^^^^ help: if this is intentional, prefix it with an underscore: `_unused`
+  |
+  = note: `#[warn(unused_variables)]` on by default",
+        ))
+        .with_stderr_does_not_contain("[..]-Zremap-path-scope=[..]")
+        .with_stderr_does_not_contain("[..]--remap-path-prefix=[..]")
+        .run();
+}
+
+#[cfg(target_os = "linux")]
+#[cargo_test(requires_readelf, nightly, reason = "-Zremap-path-scope is unstable")]
+fn object_works() {
+    use std::os::unix::ffi::OsStrExt;
+
+    let run_readelf = |path| {
+        std::process::Command::new("readelf")
+            .arg("-wi")
+            .arg(path)
+            .output()
+            .expect("readelf works")
+    };
+
+    let registry_src = paths::home().join(".cargo/registry/src");
+    let registry_src_bytes = registry_src.as_os_str().as_bytes();
+
+    Package::new("bar", "0.0.1")
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { println!("{}", file!()); }"#)
+        .publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = "0.0.1"
+           "#,
+        )
+        .file("src/main.rs", "fn main() { bar::f(); }")
+        .build();
+
+    let pkg_root = p.root();
+    let pkg_root = pkg_root.as_os_str().as_bytes();
+
+    p.cargo("build").run();
+
+    let bin_path = p.bin("foo");
+    assert!(bin_path.is_file());
+    let stdout = run_readelf(bin_path).stdout;
+    // TODO: re-enable this check when rustc bootstrap disables remapping
+    // <https://github.com/rust-lang/cargo/pull/12625#discussion_r1371714791>
+    // assert!(memchr::memmem::find(&stdout, rust_src).is_some());
+    assert!(memchr::memmem::find(&stdout, registry_src_bytes).is_some());
```

Comment:
Is there a reason we are using `memchr::memmem`?

Reply:
The output of `readelf` is huge and I am not sure about its encoding. A premature optimization maybe. Any straightforward way to do it with std?

Reply:
Oh, right, `std` doesn't have substring matching for bytes,. right?  Oh well, then its fine.